### PR TITLE
general code quality improvements

### PR DESCRIPTION
**Pull request by danieleades**
fixes a number of warnings generated by flake8 plugins not currently used in this project

These fixes address code quality, but not code *correctness*

in particular, it removes a number of blanket `# NOQA` comments that don't appear to be necessary

### CODE REVIEW

File: sphinx/ext/inheritance_diagram.py [hunk 1]
```diff
@@ -184,13 +185,11 @@ def recurse(cls: Any) -> None:
 
             # Use first line of docstring as tooltip, if available
             tooltip = None
-            try:
+            with suppress(AttributeError):  # might raise AttributeError for strange classes
                 if cls.__doc__:
                     doc = cls.__doc__.strip().split("\n")[0]
                     if doc:
                         tooltip = '"%s"' % doc.replace('"', '\\"')
-            except Exception:  # might raise AttributeError for strange classes
```
Comment (tk0miya) [inline sphinx/ext/inheritance_diagram.py:192]: Note: The contradiction between implementation and comment started from the beginning (3f7ecbde96). I don't know which is correct. But let's go with AttributeError.

**Comment by AA-Turner** [comment]
> fixes a number of warnings generated by flake8 plugins not currently used in this project

Please may you list these?

A

**Review by AA-Turner — APPROVED** [review]
In general looks OK. If we should add more things to Flake8 please do say so.

A

**Comment by danieleades** [comment]
The main ones I would normally go for are `flake8-bugbear`, `flake8-simplify`, and `flake8-comprehensions`. All of which provide fairly uncontroversial lints.

There's still a bunch of errors in this code base with these turned on though, so I can't add them in this PR without making a bunch more changes

**Comment by AA-Turner** [comment]
@danieleades you put this as draft -- I'm happy to merge but just want to check you didn't want to add anything to this?

A

**Review by AA-Turner — APPROVED** [review]

**Review by tk0miya — APPROVED** [review]

**Comment by danieleades** [comment]
> @danieleades you put this as draft -- I'm happy to merge but just want to check you didn't want to add anything to this?
> 
> A

to be honest i'm inclined to revisit and rebase this after #10602 and #10601 are merged. Those PRs add the changes in a more structured way, and update the CI to ensure they stay fixed.

**Pull request closed without merging**